### PR TITLE
PT-2018: implement pt-package-testing to tests with other products

### PR DESCRIPTION
**Pull request by eleo007**
Implement package testing of Percona-Toolkit (pt) with the other products: Percona-Server-57 (ps57), Percona-Server-80 (ps80), Percona-XtraDB-Cluster-57 (pxc57), Percona-XtraDB-Cluster-80 (pxc80), MysqL server 57 (upstream57), MySQL server 80 (upstream80), Percona Server for MongoDB 44 (psmdb44), Percona Server for MongoDB 50 (psmdb50) .

Implements two Jenkins jobs:
*  pt-package-testing : 
- performs initial setup of the node based on the distribution (installs ansible, jenkins, sets up mirrors, clones package-testing repo etc)
- runs Install, Update and "Test with another products" stages in parallel on nodes with specific distribution (bionic, centos7 etc) in AWS. Stages are performed using ansible playbooks.
- passes install_with variable to select which products should/should not be tested.

* pt-package-testing-all:
- starts parallel execution of pt-package-testing job on stretch/buster/bullseye/xenial/bionic//focal/centos7/centos8 ($node_to_test) and passes required parameters.

Example: 
1) User starts pt-package-testing-all with: 
- product_to_test = pt3
- install_repo = testing
- git_repo = default
- git_branch = default
- skip* = default
2) pt-package-testing-all in parallel:
- starts pt-package-testing job with $node_to_test = focal and parameters from user input.
- starts pt-package-testing job with $node_to_test  = bionic and parameters from user input.
- etc for every distribution.
3) pt-package-testing for $node_to_test in parallel (the same for every distribution):
- starts execution of the Install stage on focal;
- starts execution of the Update stage on focal;
- based on skip_ps80 parameter starts execution of the ps80_and_pt stage and passes install_with variable value.
- based on skip_ps57 parameter starts execution of the ps57_and_pt stage and passes install_with variable value.
- etc.

### CODE REVIEW

File: pt/jenkins/pt-package-testing-all.groovy [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                    steps {
+                        runNodeBuild('min-bullseye-x64')
+                    }
+                }
+
+                stage('Ubuntu Xenial') {
+                    steps {
+                        runNodeBuild('min-xenial-x64')
+                    }
+                }
```
Comment (EvgeniyPatlan) [inline pt/jenkins/pt-package-testing-all.groovy:119]: xenial is EOL

File: pt/jenkins/pt-package-testing.groovy [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+        sudo yum -y update
+        sudo yum install -y ansible
+    '''
+}
+
+setup_centos8_package_tests = { ->
+    sh '''
+        sed -i 's/mirror\.centos\.org/vault\.centos\.org/g;s/\#baseurl=/baseurl=/g;s/mirrorlist=/#mirrorlist=/g' /etc/yum.repos.d/CentOS*.repo
+        sudo yum install -y epel-release
+        sudo yum -y update
```
Comment (EvgeniyPatlan) [inline pt/jenkins/pt-package-testing.groovy:26]: after this step all your changes mirror->vault might be get back

File: pt/jenkins/pt-package-testing.groovy [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+    '''
+}
+
+node_setups = [
+    "min-stretch-x64": setup_stretch_package_tests,
+    "min-buster-x64": setup_buster_bullseye_package_tests,
+    "min-bullseye-x64": setup_buster_bullseye_package_tests,
+    "min-centos-7-x64": setup_centos_package_tests,
+    "min-centos-8-x64": setup_centos8_package_tests,
+    "min-xenial-x64": setup_ubuntu_package_tests,
```
Comment (EvgeniyPatlan) [inline pt/jenkins/pt-package-testing.groovy:64]: EOL

File: pt/jenkins/pt-package-testing.groovy [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+        --limit 127.0.0.1 \
+        ${playbook_path}
+
+        echo Finish: \$(date -R)
+    """
+}
+
+
+pipeline {
+    agent none
```
Comment (EvgeniyPatlan) [inline pt/jenkins/pt-package-testing.groovy:99]: never start jobs on jenkins master

File: pt/jenkins/pt-package-testing.groovy [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+        choice(
+            choices: ['pt3'],
+            description: 'Product version to test',
+            name: 'product_to_test'
+        )
+        choice(
+            choices: [
+                'min-centos-7-x64',
+                'min-centos-8-x64',
+                'min-xenial-x64',
```
Comment (EvgeniyPatlan) [inline pt/jenkins/pt-package-testing.groovy:111]: eol

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]
Also please provide an example of running job

File: pt/jenkins/pt-package-testing-all.groovy (hunk 1 above)
Comment (eleo007) [reply pt/jenkins/pt-package-testing-all.groovy]: fixed in new commit.

File: pt/jenkins/pt-package-testing.groovy (hunk 2 above)
Comment (eleo007) [reply pt/jenkins/pt-package-testing.groovy:26]: fixed in new commit.

File: pt/jenkins/pt-package-testing.groovy (hunk 3 above)
Comment (eleo007) [reply pt/jenkins/pt-package-testing.groovy]: fixed in new commit.

File: pt/jenkins/pt-package-testing.groovy (hunk 4 above)
Comment (eleo007) [reply pt/jenkins/pt-package-testing.groovy]: fixed in new commit.

**Comment by eleo007** [comment]
Hi @EvgeniyPatlan , I have made discussed changed in new commit + here is link to task execution https://pxc.cd.percona.com/job/pt-package-testing-all/45 (Some of the steps fail due to lack of space on the pxc.cd. We have discussed that we will run it on psmdb. Will move it there after the issue with running steps on the same instance is solved (Jira task is created)).

**Comment by EvgeniyPatlan** [comment]
@eleo007 please add ubuntu 22

**Review by EvgeniyPatlan — APPROVED** [review]

**Pull request merged**